### PR TITLE
SONARRUBY-1 Extract and build Ruby

### PR DESCRIPTION
**Pull request by ADarko22**

### CODE REVIEW

File: .cirrus.yml [hunk 1]
```diff
@@ -2,27 +2,121 @@ env:
...
+    folder: .gradle/caches
+  create_gradle_directory_script:
+    - mkdir -p "${CIRRUS_WORKING_DIR}/.gradle"
+
+cleanup_gradle_cache_script_template: &CLEANUP_GRADLE_CACHE_SCRIPT
+  cleanup_gradle_script:
+    - /usr/bin/find "${CIRRUS_WORKING_DIR}/.gradle/caches/" -name "*.lock" -type f -delete
+    - rm -rf "${CIRRUS_WORKING_DIR}/.gradle/caches/4.10.2/"
+    - rm -rf "${CIRRUS_WORKING_DIR}/.gradle/caches/journal-1/"
+    - rm -rf "${CIRRUS_WORKING_DIR}/.gradle/caches/build-cache-1/"
```
Comment (dorian-burihabwa-sonarsource) [inline .cirrus.yml:51]: Are we sure that the folders mentioned here are still al lrelevant?

File: .cirrus.yml [hunk 2]
```diff
@@ -2,27 +2,121 @@ env:
...
+    - build
+  <<: *ONLY_IF_SONARSOURCE_QA
+
+gradle_its_template: &GRADLE_ITS_TEMPLATE
+  <<: *SETUP_GRADLE_CACHE
+  run_its_script:
+    - |
+      if [ -n "${GIT_SUB_MODULE}" ]; then        
+        git submodule update --init --depth 1 "${GIT_SUB_MODULE}"
+      fi
```
Comment (dorian-burihabwa-sonarsource) [inline .cirrus.yml:84]: Does this need to be conditional? Can we just update the submodules whatever happens?

File: .cirrus.yml [hunk 3]
```diff
@@ -2,27 +2,121 @@ env:
...
+linux_2_cpu_4G_template: &LINUX_2_CPU_4G
+  eks_container:
+    <<: *LINUX_IMAGE
+    cpu: 2
+    memory: 4G
+
+linux_4_cpu_6G_java_17_template: &LINUX_4_CPU_6G_JAVA_17
+  eks_container:
+    <<: *LINUX_IMAGE
+    image: ${CIRRUS_AWS_ACCOUNT}.dkr.ecr.eu-central-1.amazonaws.com/base:j17-g7-latest
```
Comment (dorian-burihabwa-sonarsource) [inline .cirrus.yml:28]: Isn't this line redundant with the image line from the LINUX_IMAGE block above?
```suggestion
```

File: .cirrus.yml [hunk 4]
```diff
@@ -2,27 +2,121 @@ env:
...
+    cpu: 4
+    memory: 6G
+
 eks_container: &CONTAINER_DEFINITION
-  image: ${CIRRUS_AWS_ACCOUNT}.dkr.ecr.eu-central-1.amazonaws.com/base:latest
+  image: ${CIRRUS_AWS_ACCOUNT}.dkr.ecr.eu-central-1.amazonaws.com/base:j17-g7-latest
```
Comment (dorian-burihabwa-sonarsource) [inline .cirrus.yml:33]: it looks like this template is no longer used and can be removed

File: build.gradle [hunk 5]
```diff
@@ -0,0 +1,297 @@
...
+    }
+
+    repositories {
+        mavenLocal()
+        if (new File(rootDir, 'private/build.gradle').exists()) {
+            def repository = project.hasProperty('qa') ? 'sonarsource-qa' : 'sonarsource'
+            maven {
+                url "https://repox.jfrog.io/repox/${repository}"
+            }
+        }
```
Comment (dorian-burihabwa-sonarsource) [inline build.gradle:64]: This can probably be removed
```suggestion
```

File: build.gradle [hunk 6]
```diff
@@ -0,0 +1,297 @@
...
+
+    task javadocJar(type: Jar, dependsOn: javadoc) {
+        classifier = 'javadoc'
+        from javadoc.destinationDir
+    }
+
+    // license updater
+    spotless {
+        java {
+            targetExclude "**/src/test/resources/**", "**/build/**", "its/sources/**", "its/plugin/projects/**", "private/its/sources/**", "private/its/plugin/projects/**"
```
Comment (dorian-burihabwa-sonarsource) [inline build.gradle:161]:
```suggestion
            targetExclude "**/src/test/resources/**", "**/build/**", "its/sources/**", "its/plugin/projects/**"
```

File: build.gradle [hunk 7]
```diff
@@ -0,0 +1,297 @@
...
+
+            if (getPath().startsWith(':private')) {
+                licenseHeaderFile(rootProject.file("private/LICENSE")).updateYearWithLatest(true)
+            } else {
+                licenseHeaderFile(rootProject.file("LICENSE_HEADER")).updateYearWithLatest(true)
+            }
```
Comment (dorian-burihabwa-sonarsource) [inline build.gradle:167]:
```suggestion
            licenseHeaderFile(rootProject.file("LICENSE_HEADER")).updateYearWithLatest(true)

```

File: README.md [hunk 8]
```diff
@@ -1,3 +1,50 @@
-# sonar-ruby
+# Sonar-Go
```
Comment (dorian-burihabwa-sonarsource) [inline README.md:1]:
```suggestion
# sonar-ruby
```

**Review by dorian-burihabwa-sonarsource — CHANGES_REQUESTED** [review]
Thanks for the work, we just need a few tweaks to finish the work

File: .cirrus.yml (hunk 1 above)
Comment (ADarko22) [reply .cirrus.yml:50]: To be honest, I'm not sure, but this is present in sonar-kotlin; we can have a task for investigating it for all the extracted languages. What do you think?

File: .cirrus.yml (hunk 2 above)
Comment (ADarko22) [reply .cirrus.yml:83]: Apparently, it is required by the ruling tests only; it can be added to that task explicitly

File: .cirrus.yml (hunk 4 above)
Comment (ADarko22) [reply .cirrus.yml:32]: it should be used eventually by the promote task

**Comment by ADarko22** [comment]
The cirrus.yaml will be fixed in [SONARRUBY-2 ](https://sonarsource.atlassian.net/browse/SONARRUBY-2)

File: .cirrus.yml (hunk 1 above)
Comment (dorian-burihabwa-sonarsource) [reply .cirrus.yml:49]: Let's remember to check this later by running a build task in Cirrus ci and looking through the Gradle cache

File: .cirrus.yml (hunk 4 above)
Comment (dorian-burihabwa-sonarsource) [reply .cirrus.yml:31]: Let's see if we can simplify as we fix the pipeline in another PR

**Review by dorian-burihabwa-sonarsource — APPROVED** [review]

**Pull request merged**